command/ca/rekey.go

```diff
@@ -0,0 +1,340 @@
+'''
+
+Rekey the certificate every 16h:
+'''
+$ step ca rekey --daemon --renew-period 16h internal.crt internal.key
+'''
+
+Rekey the certificate and reload nginx:
+'''
+$ step ca rekey --daemon --exec "nginx -s reload" internal.crt internal.key
+'''
+
+Rekey the certificate and convert it to DER:
+'''
+$ step ca renew --daemon --renew-period 16h \
+  --exec "step certificate format --force --out internal.der internal.crt" \
+  internal.crt internal.key
+'''
+
+Rekey a certificate using the offline mode, requires the configuration
+files, certificates, and keys created with **step ca init**:
+'''
+$ step ca rekey --offline internal.crt internal.key
+'''
+
+Rekey the certificate and write it to specified files:
+'''
+$ step ca rekey internal.crt internal.key --out-crt foo.crt --out-key foo.key
+'''
+
+Rekey the certificate using a given private key:
+'''
+$ step ca rekey internal.crt internal.key --private-key foo.key
+'''`,
+		Flags: []cli.Flag{
+			flags.CaConfig,
+			flags.CaURL,
+			flags.Force,
+			flags.Offline,
+			flags.PasswordFile,
+			flags.Root,
+			cli.StringFlag{
+				Name: "expires-in",
+				Usage: `The amount of time remaining before certificate expiration,
+at which point a renewal should be attempted. The certificate renewal will not
+be performed if the time to expiration is greater than the **--expires-in** value.
+A random jitter (duration/20) will be added to avoid multiple services hitting the
+renew endpoint at the same time. The <duration> is a sequence of decimal numbers,
+each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m".
+Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".`,
+			},
+			cli.IntFlag{
+				Name: "pid",
+				Usage: `The process id to signal after the certificate has been renewed. By default the
+the SIGHUP (1) signal will be used, but this can be configured with the **--signal**
+flag.`,
+			},
+			cli.StringFlag{
+				Name: "pid-file",
+				Usage: `The <file> from which to read the process id that will be signaled after the certificate
+has been renewed. By default the the SIGHUP (1) signal will be used, but this can be configured with the **--signal**
+flag.`,
+			},
+			cli.IntFlag{
+				Name: "signal",
+				Usage: `The signal <number> to send to the selected PID, so it can reload the
+configuration and load the new certificate. Default value is SIGHUP (1)`,
+				Value: int(syscall.SIGHUP),
+			},
+			cli.StringFlag{
+				Name:  "exec",
+				Usage: "The <command> to run after the certificate has been renewed.",
+			},
+			cli.BoolFlag{
+				Name: "daemon",
+				Usage: `Run the renew command as a daemon, renewing and overwriting the certificate
+periodically. By default the daemon will renew a certificate before 2/3 of the
+time to expiration has elapsed. The period can be configured using the
+**--renew-period** or **--expires-in** flags.`,
+			},
+			cli.StringFlag{
+				Name: "renew-period",
```

Comment:
@dopey should the flag change to rekey-period instead?

Reply:
yep! I'm actually posting a few changed right now, so I can add that to my changes.